apps/web/src/App.tsx

```diff
@@ -59,80 +59,88 @@ function App() {
       <QueryClientProvider client={queryClient}>
         <AuthHandlerComponent>
           <ThemeHandlerComponent>
-            <Switch>
-              <Route path="/auth/signup">
-                <SignUpPage />
-              </Route>
-              <Route path="/auth/login">
-                <LoginPage />
-              </Route>
-              <Route path="/auth/reset/request">
-                <PasswordResetPage />
-              </Route>
-              <Route path="/auth/reset/:token">
-                <PasswordResetPage />
-              </Route>
-              <Route path="/auth/invitation/:token">
-                <InvitationScreen />
-              </Route>
-              <Route path="/">
-                <PrivateRoute>
-                  <Switch>
-                    <Route exact path="/onboarding/application">
+            <Routes>
+              <Route path="/auth/signup" element={<SignUpPage />} />
+              <Route path="/auth/login" element={<LoginPage />} />
+              <Route path="/auth/reset/request" element={<PasswordResetPage />} />
+              <Route path="/auth/reset/:token" element={<PasswordResetPage />} />
+              <Route path="/auth/invitation/:token" element={<InvitationScreen />} />
+              <Route element={<AppLayout />}>
+                <Route
+                  path="/*"
+                  element={
+                    <RequiredAuth>
+                      <HomePage />
+                    </RequiredAuth>
+                  }
+                />
+                <Route
+                  path="/onboarding/application/*"
+                  element={
+                    <RequiredAuth>
                       <ApplicationOnBoarding />
-                    </Route>
-                    <AppLayout>
-                      <Route exact path="/">
-                        <HomePage />
-                      </Route>
-                      <Route exact path="/templates/create">
-                        <TemplateEditorPage />
-                      </Route>
-                      <Route exact path="/templates/edit/:templateId">
-                        <TemplateEditorPage />
-                      </Route>
-                      <Route exact path="/templates">
-                        <NotificationList />
-                      </Route>
-                      <Route exact path="/activities">
-                        <ActivitiesPage />
-                      </Route>
-                      <Route exact path="/settings/widget">
-                        <WidgetSettingsPage />
-                      </Route>
-                      <Route exact path="/settings/organization">
-                        <OrganizationSettingsPage />
-                      </Route>
-                    </AppLayout>
-                  </Switch>
-                </PrivateRoute>
+                    </RequiredAuth>
+                  }
+                />
+                <Route
+                  path="/templates/create/*"
```

Comment:
Not familiar with the new react router syntax, is the `/*` in the end mandatory?

Reply:
According to the documentation...
`<Route exact>` is gone. Instead, routes with descendant routes (defined in other components) use a trailing `*` in their path to indicate they match deeply

Reply:
And in case we don't need a deep match but exact match can we omit here the '/*' ?
```suggestion
                  path="/templates/create"
```